connection.go

```diff
@@ -73,9 +69,18 @@ func (sc *snowflakeConn) exec(
 	req := execRequest{
 		SQLText:    query,
 		AsyncExec:  noResult,
+		IsInternal: isInternal,
 		SequenceID: counter,
 	}
-	req.IsInternal = isInternal
+	params := []paramKey{MultiStatementCount}
+	for _, k := range params {
+		key := ctx.Value(k)
+		if key != nil {
+			req.Parameters = map[string]interface{}{string(MultiStatementCount): key}
+		}
+	}
```

Comment:
I can see what the for loop is trying to do (save on some typing later via the params slice), but this doesn't seem necessary. The following is much clearer albeit more boilerplate-y:

```
req := execRequest{
  // ...
  Parameters: map[string]interface{}{},
  // ...
}

if key := ctx.Value(MultiStatementCount); key != nil {
  req.Parameters[string(MultiStatementCount)] = key
}
if key := ctx.Value(SomeOtherThing); key != nil {
  req.Parameters[string(SomeOtherThing)] = key
}
if key := ctx.Value(YetAnotherThing); key != nil {
  req.Parameters[string(YetAnotherThing)] = key
}
```